### PR TITLE
samples: bluetooth: hci_uart: Improve TX path

### LINKED ISSUE #2955: Use interrupt-driven TX in hci_uart sample
**_Reported by Carles Cufi:_**

The current hci_uart sample uses uart_poll_out() to send bytes out to the Host. It should instead use uart_fifo_fill() and wait on a semaphore until there's space in the UART FIFO to send some more if it's full.

(Imported from Jira ZEP-1485)

### PR DESCRIPTION
**Pull request by nordic-krch**
Improved TX path to use uart_fifo_fill instead of uart_poll_out

PR takes TX part from #27581 with a bit of rework. It is using very similar approach as in `h4.c`.

Fixes #2955

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
This looks reasonable to me, thank you for implementing this

**Comment by carlescufi** [comment]
I also tested this with BlueZ without any issues.

**Comment by nordic-krch** [comment]
my test setup consist of two nrf52 boards, one hci_uart and second peripheral_hr and controller off.

**Review by joerchan — APPROVED** [review]
LGMT, please consider using suggested pointer check for consistency.

**Review by anangl — APPROVED** [review]

File: samples/bluetooth/hci_uart/src/main.c [hunk 1]
```diff
@@ -110,6 +113,27 @@ static void h4_acl_recv(struct net_buf *buf, int *remaining)
 	LOG_DBG("len %u", *remaining);
 }
 
+static void tx_isr(void)
+{
+	static struct net_buf *buf;
+	int len;
+
+	if (buf == NULL) {
```
[the remark this answers is not in the document]
Comment (nordic-krch) [reply samples/bluetooth/hci_uart/src/main.c]: done

**Pull request merged**